doc/vxlan/Vxlan_hld.md

```diff
@@ -0,0 +1,338 @@
+| VM                       | Virtual Machine                |
+| VRF                      | Virtual Routing and Forwarding |
+| VNet                     | Virtual Network                |
+
+# 1 Requirements Overview
+## 1.1 Functional requirements
+This section describes the SONiC requirements for Vxlan feature. 
+
+At a high level the following should be supported:
+
+Phase #1
+- Should be able to perform the role of Vxlan Tunnel End Point (VTEP)
+- VNet peering between customer VMs and Baremetal servers [VNet Requirements](https://github.com/lguohan/SAI/blob/vni/doc/SAI-Proposal-QinQ-VXLAN.md).
+- Distributed Vxlan routing with Symmetric IRB model (RIOT)
+
+Phase #2
+- Integration with BGP EVPN and create VRF devices in linux kernel
+- Should support untagged or tagged traffic (Overlay layer 2 networks over layer 3 underlay)
+- Should be able to do HER for unicast traffic based on configured flood list
+- CLI commands to configure Vxlan
+
+
+## 1.2 Orchagent requirements
+### Vxlan orchagent:
+ - Should be able to create VRF/VLAN to VNI mapping and also track peering configurations. 
+ - Should be able to create VRFs per VNET tables. 
+ - Should be able to create tunnels and encap/decap mappers. 
+
+### Route orchagent:
+ - Should be able to handle routes within a VNet 
+ - Should be able to create NH tunnels for the endpoints
+ - Should be VNet/VRF aware
+
+### FDB orchagent:
+ - Should be VTEP aware
+ - Should support static configuration of FDB entries learnt on remote VTEP
+ 
+### INTFs orchagent:
+ - Should be VNet/VRF aware
+ 
+## 1.3 CLI requirements
+- User should be able to get FDB learnt per VNI
+- User should be able to configure Vxlan tunnels and VTEPs (Overlay)
+
+In summary:
+```
+	- config vxlan <vxlan_name>
+	- config vxlan <vxlan_name> vlan <vlan_id> vni <vni_id>
+	- config vxlan <vxlan_name> src_lb_if <loopback_if>
+	- config vxlan <vxlan_name> vlan <vlan_id> flood vtep <ip1, ip2, ip3>
+	- show mac <vxlan_name>
+	- show vxlan <vxlan_name>
+```
+Configuring VNet peering via CLI is beyond the scope
+
+## 1.4 Scalability requirements
+
+### 2.1.1 VNet Peering
+###### Table 2: VNet peering scalability
+| Vxlan component          | Expected value              |
+|--------------------------|-----------------------------|
+| VNI                      | 8k                          |
+| Tunnel encaps            | 128k                        |
+| VMs                      | 512k                        |
+| VRFs                     | 128                         |
+
+# 2 Modules Design
+
+## 2.1 Config DB
+Following new tables will be added to Config DB
+### 2.1.1 VXLAN Table
+
+
+```
+VXLAN_TUNNEL:{{tunnel_name}} 
+    "src_ip": {{ip_address}} 
+    "dst_ip": {{ip_address}}
+
+VXLAN_TUNNEL_MAP|{{tunnel_name}}|{{tunnel_map}}
+    "vni_id": {{ vni_id}}
+    "vlan_id": {{ vlan_id }}
+```
+### 2.1.2 VNET Table
+```
+VNET:{{vnet_name}} 
+    "vxlan_tunnel": {{tunnel_name}}
+    "vxlanid": {{vni}} 
+    "peer_list": {{vnet_name_list}} 
```

Comment:
Probable it's better to have one name of the attribute to refer to vni? I'd put `vni` here.